### ISSUE TITLE
Background healing newer finish when changing server by new.

### ISSUE THREAD

**Issue by pulchart** [minio/minio#16666]
Hello

I'm observing strange behaviour of background healing when changing server fleet of minio service. The background healing newer finish for us and I need to force it by some manual "tricks" (restarting services repeatedly, running multiple time "mc admin heal -r alias").

Can you please check what is wrong with the healing in minio? I can easily reproduce it and run test as needed.

## Expected Behavior
* Change server by new with empty disks,
* background healing is running by "constant speed" and finish without manual operation tasks.

## Current Behavior
I would like to show it of attached "[healing.png](https://drive.google.com/file/d/19ro1nQh6MxICRKZ6WjiieJ5peVkUtgoD/view?usp=sharing)" graph. The graph reports how the disk are filled so it shows how healing is working in time:
0 - The healing is expected to finish when disks are having ~2TiB of free space.
1 - One server is replaced with new disks in each Minio pool (the disk are empty and havin ~4 TiB free)
2 - Background healing starts for almost all disks == OK
3 - Few disks do not start healing == WRONG
4 - The haling stops at ~3TiB for all disks == WRONG
now I need to manually restart the minio service  ```minio admin service alias restart```
5 - The few disks from (step 3) start to heal, but not all (disks from step4) == WRONG
after few hours later it stops the helaing of all disks again, the space free us still about ~3TiB.
6 - I have to run ```minio admin heal -r alias``` to enforce healing.

## Steps to Reproduce (for bugs)
1. Replace minio server in a pool by new with empty disks. 
2. Wait and observe storage usage by healing process. It shows "Estimated: 1 day" for few hours.
3. The healing stops to fill disks after few hours and the estimated healing time of disks start to grow like to days at first, then grows to months.

## Context

How has this issue affected you?
* Healings of new disks newer finish automatically. We must run manual restarts of service and run foreground healing process repeatedly (manually).

## Your Environment

* Version used 
minio version RELEASE.2023-02-17T17-52-43Z (commit-id=84bb7d05a998a096b96acd68ea94053d28a87eb3)
Runtime: go1.19.6 linux/amd64
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Copyright: 2015-2023 MinIO, Inc.

* Server setup:
Two pools, each pool of 16 servers, each server 4 disks.

* Operating System and version:
CentOS 9 Stream

**Comment by vadmeste** [comment]
@pulchart does manual healing successfully finish ? can you upload all logs that you have ?

**Comment by pulchart** [comment]
> @pulchart does manual healing successfully finish ?

Yes the manual healing was successful. Example of one last run see here [one_from_the_loop.png](https://drive.google.com/file/d/1VDrSBT7sVQdo84jck1mzuSJdhauNrevv/view?usp=sharing) 

> can you upload all logs that you have ?

They are full of the manual restarts / updates etc... as I try to trigger the healing. Are you interested in some period of the time from healing or after some action?

**Comment by pulchart** [comment]
Hello, I replaced more servers from each pool and observed the healing.

**For context:**

We can replace up to 16 disks (up to 4x servers) per each pool by new ones at any time in parallel as we have 16x servers per each pool and each server has 4x disks.

My initial report shows healing for case when 1x server is replaced (with new dicks) from each pool. 

**Now:**

I replaced +2x servers from each pool and the healing is now running at 3x servers from each pool (12x disks in each pool) in parallel.

**Report:**

See sections in linked png graph [slow_healing.png](https://drive.google.com/file/d/1ABNSkcMcV5jvl8m2xNtgN-G4gNpz-XBU). It shows space usage going down by healing process. I mark there five sections:

Section 1) my original report

Section 2) replaced +2x server in each pool

Background healing was going well for pool 2 disks but slow for pool 1 disks. However it slow down for pool 2 after +7h and not mowing forward.

Section 3) I restart the service ```mc admin service alias restart```

Healing was going well for pool 1, but just for few minutes. Both pools healing is super slow again.

Section 4) I run foreground healing ```mc admin heal -r -v alias```

Foreground healing was going well for pool 1 but was slow for pool 2. It slowdown after +7h again for all disks. **The healing scanned just ~3k object from ~16M in ~2 hours**.

Section 5) I run multiple ```mc admin heal -r -v alias/pathXY```

I kill previous ```mc admin heal -r -v alias``` and instead of it I parallelized it by command like this:
```
mc admin heal -r -v alias/<path1>
mc admin heal -r -v alias/<path2>
mc admin heal -r -v alias/<path3>
...
mc admin heal -r -v alias/<path15>
```

**Healing sped up however again pool by pool. Not in parallel,**
